### PR TITLE
[Release] Enable tiling oriented detection for v0.4.0/geti1.1.0

### PR DESCRIPTION
**Pull request by eugene123tw**

### CODE REVIEW

**Comment by eugene123tw** [comment]
run ote-test

**Comment by eugene123tw** [comment]
https://ci-ote.iotg.sclab.intel.com/job/ote/job/pr-ote-test/826/

File: ote_sdk/ote_sdk/usecases/exportable_code/demo/demo_package/model_container.py [hunk 1]
```diff
@@ -110,7 +109,10 @@ def infer(self, frame):
         """
         # getting result include preprocessing, infer, postprocessing for sync infer
         predictions, frame_meta = self.core_model(frame)
-        predictions = detection2array(predictions)
+
+        # MaskRCNN returns tuple so no need to process
+        if not self.segm:
+            predictions = detection2array(predictions)
```
Comment (harimkang) [inline ote_sdk/ote_sdk/usecases/exportable_code/demo/demo_package/model_container.py:115]: As I analyzed yesterday, Classification also passes through the code and causes an issue here. It seems to me that this code only filters out segmentation, is that correct?

File: external/model-preparation-algorithm/configs/rotated-detection/efficientnetb2b_maskrcnn/data_pipeline.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 dataset_type = "CocoDataset"
 img_size = (1024, 1024)
 
-img_norm_cfg = dict(mean=(103.53, 116.28, 123.675), std=(1.0, 1.0, 1.0), to_rgb=False)
+img_norm_cfg = dict(mean=(103.53, 116.28, 123.675), std=(1.0, 1.0, 1.0), to_rgb=True)
```
Comment (harimkang) [inline external/model-preparation-algorithm/configs/rotated-detection/efficientnetb2b_maskrcnn/data_pipeline.py:4]: @jaegukhyun Could you please comment on this? Any problem?

**Comment by harimkang** [comment]
> https://ci-ote.iotg.sclab.intel.com/job/ote/job/pr-ote-test/826/

As confirmed with this build, there are still issues with classification and segmentation.

File: ote_sdk/ote_sdk/usecases/exportable_code/demo/demo_package/model_container.py [hunk 3]
```diff
@@ -110,7 +109,10 @@ def infer(self, frame):
         """
         # getting result include preprocessing, infer, postprocessing for sync infer
         predictions, frame_meta = self.core_model(frame)
-        predictions = detection2array(predictions)
+
+        # MaskRCNN returns tuple so no need to process
+        if not self.segm:
```
Comment (goodsong81) [inline ote_sdk/ote_sdk/usecases/exportable_code/demo/demo_package/model_container.py]:
```suggestion
        if self._task_type == TaskType.DETECTION:
```

If I'm correctly understanding, detection2array() is only for detection task.

**Review by goodsong81 — CHANGES_REQUESTED** [review]
Need to resolve test failures.

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
https://ci-ote.iotg.sclab.intel.com/job/ote/job/pr-ote-test/828/

**Comment by goodsong81** [comment]
Geti enabling test (for reference): https://ci.iotg.sclab.intel.com/job/IMPT/job/IMPTOps/48653/

**Review by harimkang — APPROVED** [review]
image
all passed ote cli's mpa test in my local

**Comment by eugene123tw** [comment]
Passed: 1170 (88.77%)
Ignored: 147 (11.15%)
Failed: 1 (0.08%)


@harimkang @goodsong81 failed case is related to anomaly seg accuracy assertion. 
```
template = ModelTemplate(model_template_id='ote_anomaly_segmentation_stfpm', model_template_path='/usr/src/app/ote_validation/tra...able_code_paths=ExportableCodePaths(default=None, openvino=None), task_type_sort_priority=-1, gigaflops=5.6, size=21.1)
root = '/tmp/ote_cli/'
ote_dir = '/usr/src/app/ote_validation/training_extensions'
args = {'--input': 'data/anomaly/shapes/test/hexagon', '--test-ann-files': 'data/anomaly/segmentation/test.json', '--test-data-roots': 'data/anomaly/shapes', '--train-ann-file': 'data/anomaly/segmentation/train.json', ...}
threshold = 0.01

    def ote_eval_openvino_testing(template, root, ote_dir, args, threshold):
        work_dir, template_work_dir, _ = get_some_vars(template, root)
        command_line = [
            "ote",
            "eval",
            template.model_template_path,
            "--test-ann-file",
            f'{os.path.join(ote_dir, args["--test-ann-files"])}',
            "--test-data-roots",
            f'{os.path.join(ote_dir, args["--test-data-roots"])}',
            "--load-weights",
            f"{template_work_dir}/exported_{template.model_template_id}/openvino.xml",
            "--save-performance",
            f"{template_work_dir}/exported_{template.model_template_id}/performance.json",
        ]
        assert run(command_line, env=collect_env_vars(work_dir)).returncode == 0
        assert os.path.exists(
            f"{template_work_dir}/exported_{template.model_template_id}/performance.json"
        )
        with open(
            f"{template_work_dir}/trained_{template.model_template_id}/performance.json"
        ) as read_file:
            trained_performance = json.load(read_file)
        with open(
            f"{template_work_dir}/exported_{template.model_template_id}/performance.json"
        ) as read_file:
            exported_performance = json.load(read_file)
    
        for k in trained_performance.keys():
>           assert (
                exported_performance[k] > trained_performance[k]
                or abs(trained_performance[k] - exported_performance[k])
                / (trained_performance[k] + 1e-10)
                <= threshold
            ), f"{trained_performance[k]=}, {exported_performance[k]=}"
E           AssertionError: trained_performance[k]=0.29357798165137616, exported_performance[k]=0.28203891801085723

ote_cli/ote_cli/utils/tests.py:215: AssertionError
```

**Review by goodsong81 — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Pull request merged**